pkg/operator/controllers/gameserverbuild_controller.go

```diff
@@ -177,25 +199,37 @@ func (r *GameServerBuildReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 			GameServersUnhealthyCounter.WithLabelValues(gsb.Name).Inc()
 			r.expectations.addGameServerToUnderDeletionMap(gsb.Name, gs.Name)
 			r.Recorder.Eventf(&gsb, corev1.EventTypeNormal, "Crashed", "GameServer %s was deleted because it crashed, state: %s, health: %s", gs.Name, gs.Status.State, gs.Status.Health)
+
+			pendingCleanUpCount++
+			timeToDeleteBySum += math.Abs(float64(time.Until(gs.DeletionTimestamp.Time).Milliseconds()))
+		}
+		if gs.Status.State != gs.Status.PrevState {
+			gs.Status.PrevState = gs.Status.State
 		}
 	}
 
+	if recentStandingByCount > 0 {
+		GameServersCreatedDuration.WithLabelValues(gsb.Name).Set(timeToStandBySum / float64(recentStandingByCount))
+	}
+
+	if pendingCleanUpCount > 0 {
+		GameServersCleanUpDuration.WithLabelValues(gsb.Name).Set(timeToDeleteBySum / float64(pendingCleanUpCount))
+	}
+
 	// calculate the total amount of servers not in the active state
 	nonActiveGameServersCount := standingByCount + initializingCount + pendingCount
 
 	// user has decreased standingBy numbers
 	if nonActiveGameServersCount > gsb.Spec.StandingBy {
-		totalNumberOfGameServersToDelete := int(math.Min(float64(nonActiveGameServersCount-gsb.Spec.StandingBy), maxNumberOfGameServersToDelete))
-		err := r.deleteNonActiveGameServers(ctx, &gsb, &gameServers, totalNumberOfGameServersToDelete)
-		if err != nil {
-			return ctrl.Result{}, err
+		var totalNumberOfGameServersToDelete int
+
+		// we also need to check if we are above the max
+		// this can happen if the user modifies the spec.Max during the GameServerBuild's lifetime
+		if nonActiveGameServersCount+activeCount > gsb.Spec.Max {
+			totalNumberOfGameServersToDelete += int(math.Min(float64(nonActiveGameServersCount+activeCount-gsb.Spec.Max), maxNumberOfGameServersToDelete))
 		}
-	}
 
-	// we need to check if we are above the max
-	// this can happen if the user modifies the spec.Max during the GameServerBuild's lifetime
-	if nonActiveGameServersCount+activeCount > gsb.Spec.Max {
-		totalNumberOfGameServersToDelete := int(math.Min(float64(nonActiveGameServersCount+activeCount-gsb.Spec.Max), maxNumberOfGameServersToDelete))
+		totalNumberOfGameServersToDelete = int(math.Min(float64(totalNumberOfGameServersToDelete+nonActiveGameServersCount-gsb.Spec.StandingBy), maxNumberOfGameServersToDelete))
```

Comment:
Consolidated these two "non-active" scalers. Less bulk, easier to apply consistent metrics to.